bin/codemods/src/assign-component-to-displayname-variable.js

```diff
@@ -0,0 +1,160 @@
+    const replaceClassOrGetValue = displayName => arg =>
+        argIsComponent( arg )
+            ? j.identifier( displayName )
+            : arg.value;
+
+    const exportDefaultDeclarations = root
+        .find( j.ExportDefaultDeclaration );
+
+    exportDefaultDeclarations.forEach( node => {
+        const calleeNameValue = node.get( 'declaration', 'callee', 'name' ).value
+
+        if ( ! calleeNameValue ) {
+            // return early, no immediate function call
+            return;
+        }
+
+        const exportDefaultInstance = j( node );
+
+        const args = node.get( 'declaration', 'arguments' );
+        const matchingArg = getMatchingArg( args );
+
+        if ( ! matchingArg ) {
+            // return early, no createClass or extends Component argument found
+            return;
+        }
+
+        const isCreateClassInstance = argIsCreateClassInstance( matchingArg )
+
+        const propertyType = isCreateClassInstance
+            ? j.Property
+            : j.ClassProperty;
+
+        const classExpressions = isCreateClassInstance
+            ? exportDefaultInstance.find( j.CallExpression, createClassIdentifier )
+            : exportDefaultInstance.find( j.ClassExpression )
+
+        const displayNameValue = classExpressions
+            .find( propertyType, displayNamePropertyIdentifier )
+            .at( 0 )
+            .get( 'value', 'value' )
+            .value;
+
+        if ( ! displayNameValue ) {
+            // return early, no displayNameValue
+            return;
+        }
+
+        exportDefaultInstance.replaceWith(
+            node => [
+                ( isCreateClassInstance
+                    // if createClass, replace with const x = React.createClass...
+                    ? (
+                        j.variableDeclaration( 'const', [
+                            j.variableDeclarator(
+                                j.identifier( displayNameValue ),
+                                matchingArg.value
+                            )
+                        ] )
+                    )
+                    // else Replace with class x extends {original-super-class}...
+                    : (
+                        j.classDeclaration(
+                            j.identifier( displayNameValue ),
+                            matchingArg.value.body,
+                            matchingArg.value.superClass
+                        )
+                    )
+                ),
+                // In both cases export with export default calleeName( x ),
+                // maintaining argument position
+                j.exportDefaultDeclaration(
+                    j.callExpression(
+                        j.identifier( calleeNameValue ),
+                        args.map( replaceClassOrGetValue( displayNameValue ) )
+                    )
+                ),
+            ]
+        );
+    } );
+
+    return exportDefaultDeclarations.toSource( {
+        useTabs: true,
+    } );
+}
```

Comment:
i might be wrong, but I think this whole file is currently being indented with spaces instead of tabs.

can we change it to tabs instead? (and if you want autoformatting feel free to prepend the file with: `/** @format */`

Reply:
Oops! Bit of an oversight on my part! All fixed up now thanks to prettier :)